test/example/typedef.txt

```diff
@@ -0,0 +1,18 @@
+# Signature
+(func (param i32))
+(ref (func (param i32)))
+(ref null (func (param i32)))
+# Struct
+(struct (field i32 i64 (mut f32) (mut f64) externref))
+(ref (struct (field i32 i64 (mut f32) (mut f64) externref)))
+(ref null (struct (field i32 i64 (mut f32) (mut f64) externref)))
```

Comment:
Note that horrible things will happen currently if these are cyclic. In text format, subtypes would rather be referenced by type section index or name, but that's not applicable to printing out of module context. What can be done now I guess is to keep a set of types already seen, and emit a non-standard `(cyclic)`?

Reply:
I think we should totally ignore recursive types for now and leave them to follow up work. IIUC, it's not even possible to construct a cyclic type in the current implementation, right?

Reply:
Hmm, good question. Would assume that cyclic types are possible, at least I don't see how one would represent tree-likes otherwise. Given that the printing isn't used anyway yet, leaving this for a future PR should be fine, though.

Reply:
Ah, perhaps we mean different things by cyclic? Was your original comment just about deeply nested constructed types as opposed to recursive types (i.e. types that contain themselves)? There's currently no way to construct recursive types because all the components of a type have to exist already at the time of the type's creation, but we should certainly be able to construct arbitrarily nested types. For now I'd say we should just print the whole thing and we can consider various ways to make the printing more ergonomic as a follow-on.